fastscanner/adapters/rest/scanner.py

```diff
@@ -118,3 +134,44 @@ async def websocket_realtime_scanner(
         if scanner_id:
             await service.unsubscribe_realtime(scanner_id)
             logger.info(f"Unsubscribed scanner {scanner_id}")
+
+
+@router.post("/scan")
+async def scan(
+    request: ScanRequest, service: ScannerService = Depends(get_scanner_service)
+) -> ScanResponse:
+    try:
+        ClockRegistry.set(LocalClock())
+        scanner = ScannersLibrary.instance().get(request.type, request.params)
+
+        symbols = await service._symbols_provider.active_symbols()
+        all_results = []
+
+        scan_tasks = [
+            _scan_symbol(scanner, symbol, request.start, request.end, request.freq)
+            for symbol in symbols
+        ]
+
+        scan_results = await asyncio.gather(*scan_tasks, return_exceptions=True)
+
+        for symbol, result in zip(symbols, scan_results):
+            if isinstance(result, pd.DataFrame) and not result.empty:
+                logger.info(result)
+                result = result.reset_index()
+                result["symbols"] = symbol
+                symbol_results = result.to_dict(orient="records", index=True)
+                all_results.extend(symbol_results)
```

Comment:
Let's implement instead a scan_all method in the service and move all this logic there. The rest adapter needs to handle only serialization/deserialization operations and IO.